app/main.py

```diff
@@ -1,34 +1,49 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int = 10,
+                 column: int = 10) -> None:
+        self.row = row
+        self.column = column
 
+    def create_field(self) -> list:
+        matrix = [[u"\u007E" for _ in range(self.column)]
+                  for _ in range(self.row)]
+        return matrix
 
-class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+class Ship:
+    def __init__(self, ships: list) -> None:
+        self.ship = ships
+        self.matrix = Deck().create_field()
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self) -> list:
+        for i in range(len(self.ship)):
+            cordinats_1 = self.ship[i][0]
+            cordinats_2 = self.ship[i][1]
+            row_1, column_1 = cordinats_1
+            row_2, column_2 = cordinats_2
+            if row_1 == row_2:
+                for elem in range(column_1, column_2 + 1):
+                    self.matrix[row_1][elem] = u"\u25A1"
+            if column_1 == column_2:
+                for elem in range(row_1, row_2 + 1):
+                    self.matrix[elem][column_1] = u"\u25A1"
+        return self.matrix
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
+    def __init__(self, ships: list) -> None:
+        self.ship = ships
+        self.matrix = Ship(ships).get_deck()
 
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def fire(self, kord: tuple) -> str:
+        x, y = kord
+        if self.matrix[x][y] == u"\u25A1":
+            self.matrix[x][y] = "*"
+            for i in self.ship:
+                a, b = i
+                if (self.matrix[a[0]][a[1]] == "*"
+                        and self.matrix[b[0]][b[1]] == "*"):
```

Comment:
The logic for checking if a ship is sunk assumes that the ship is represented by exactly two coordinates. If ships can be of different lengths, this logic will not work correctly. Consider iterating over the range of coordinates that represent the ship to check if all parts are hit.